lib/client/doc.js

```diff
@@ -58,6 +68,8 @@ function Doc(connection, collection, id) {
   this.type = null;
   this.data = undefined;
 
+  this._docPresence = new connection.DocPresence(this);
```

Comment:
As I've mentioned above, `Doc` won't have access to `connection.DocPresence` the way this is currently put together, because `connection.DocPresence` is set by `Backend`, which might be on a different machine altogether.